### ISSUE TITLE
User function Snapshot and Event formats

### ISSUE THREAD

**Issue by viktorklang** [cloudstateio/cloudstate#25]
We need to document somewhere as to what data format these should be in. Protobuf is not the ideal data storage format as it is designed for protocols and not data storage. Should we mandate a format or leave it up to the users?

**Comment by marcellanz** [comment]
I'm interested in this from the "Language "X" and its Client API perspective. 

The format used is transparent for the proxy, isn't it?

A Python client may [_pickle_](https://docs.python.org/3.7/library/pickle.html) its events; JS probably use JSON and Go use [_gob_](https://golang.org/pkg/encoding/gob/) or _protobufs_... or any other preferred format. This might vary if "Function Code" implementations can be mixed at the same time or one expects to have snapshots persisted over a long time and Function Code has expectations about persisted data to be migrate-able over time.

**Comment by viktorklang** [comment]
@marcellanz Yes, the format should be opaque to the proxy, but of course this has implications over time for maintenance, interoperability if one wants to rewrite a function from one language to another.

**Comment by marcellanz** [comment]
How much of this issue is "answered" by 46d1dc8f01d9f9413a4ca8c1d908b041906007c8 and [CloudState serialization convention](https://cloudstate.io/docs/developer/language-support/serialization.html)? The later explains the conventions how a different format should be implemented for CloudState.

Perhaps, it could be explicitly documented why the convention is relevant even if opaque to the proxy and why "stability of the encoded form" is important in this case.
